### PR TITLE
Helios gateway rename

### PR DESCRIPTION
**Pull request by nmonterroso**
@Wikia/services-team 

Renaming `:lib:service-gateway:helios-gateway` to `:lib:service-gateway:authentication-gateway` and removing subproject with swagger-generated client. Built on top of Wikia/pandora#612.

### CODE REVIEW

File: service/external-auth/src/main/java/com/wikia/services/externalauth/ExternalAuthApplication.java [hunk 1]
```diff
@@ -33,7 +33,7 @@ public void initialize(Bootstrap<ExternalAuthConfiguration> bootstrap) {
         ExternalAuthApplication.class,
         (builder -> {
```
Comment (WikiaTech) [inline service/external-auth/src/main/java/com/wikia/services/externalauth/ExternalAuthApplication.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: service/external-auth/src/test/java/com/wikia/services/externalauth/helpers/TestHelper.java [hunk 2]
```diff
@@ -1,6 +1,6 @@
 package com.wikia.services.externalauth.helpers;
 
-import com.wikia.gateway.helios.domain.AccessData;
+import com.wikia.services.authentication.gateway.domain.AccessData;
 import com.wikia.services.externalauth.domain.LinkedGoogleAccount;
 
 public class TestHelper {
```
Comment (WikiaTech) [inline service/external-auth/src/test/java/com/wikia/services/externalauth/helpers/TestHelper.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1118)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

File: service/external-auth/src/test/java/com/wikia/services/externalauth/helpers/TestHelper.java (hunk 2 above)
Comment (nmonterroso) [inline service/external-auth/src/test/java/com/wikia/services/externalauth/helpers/TestHelper.java]: This is taken care of in Wikia/pandora#610

File: lib/service-gateway/external-auth-gateway/src/main/java/com/wikia/services/externalauth/gateway/ExternalAuthGatewayImpl.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      ServiceProvider serviceProvider,
+      @ServiceEnvironment String environment) {
+
+    this.serviceProvider = serviceProvider;
+    this.environment = environment;
+  }
+
+  @Override
+  public GoogleApi googleApi() throws GatewayException {
+    return new GoogleApi(getApiClient());
```
Comment (sqreek) [inline lib/service-gateway/external-auth-gateway/src/main/java/com/wikia/services/externalauth/gateway/ExternalAuthGatewayImpl.java]: In my opion is still bad pattern to create wrapper inside wrapper ... getApiClient logic should be merged into googleApi()

**Comment by sqreek** [comment]
Lgtm

File: lib/service-gateway/external-auth-gateway/src/main/java/com/wikia/services/externalauth/gateway/ExternalAuthGatewayImpl.java (hunk 3 above)
Comment (nmonterroso) [inline lib/service-gateway/external-auth-gateway/src/main/java/com/wikia/services/externalauth/gateway/ExternalAuthGatewayImpl.java]: I disagree. The *Api classes are generated from the tags in the swagger spec, and each spec can have any number of tags. Helios has 5 tags, so 5 *Api classes would be generated. Also here, each external service would be its own Api class (Facebook). I'd rather have a helper method to do that than to repeat the code.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Pull request merged**